dexcom/calibration.go

```diff
@@ -103,23 +114,35 @@ func NewCalibration() *Calibration {
 }
 
 func (c *Calibration) Parse(parser structure.ObjectParser) {
+	c.ID = parser.String("recordId")
 	c.SystemTime = TimeFromRaw(parser.Time("systemTime", TimeFormat))
 	c.DisplayTime = TimeFromRaw(parser.Time("displayTime", TimeFormat))
 	c.Unit = parser.String("unit")
 	c.Value = parser.Float64("value")
 	c.TransmitterID = parser.String("transmitterId")
+	c.TransmitterGeneration = parser.String("transmitterGeneration")
+	c.DisplayDevice = parser.String("displayDevice")
+	c.TransmitterTicks = parser.Int("transmitterTicks")
 }
 
 func (c *Calibration) Validate(validator structure.Validator) {
 	validator = validator.WithMeta(c)
+	validator.String("recordId", c.ID).Exists().NotEmpty()
 	validator.Time("systemTime", c.SystemTime.Raw()).Exists().NotZero().BeforeNow(SystemTimeNowThreshold)
 	validator.Time("displayTime", c.DisplayTime.Raw()).Exists().NotZero()
-	validator.String("unit", c.Unit).Exists().OneOf(CalibrationUnits()...)
+	validator.String("transmitterId", c.TransmitterID).Exists().Using(TransmitterIDValidator)
+	validator.Int("transmitterTicks", c.TransmitterTicks).Exists().GreaterThan(0)
+	validator.String("displayDevice", c.DisplayDevice).Exists().NotEmpty()
+	validator.String("transmitterGeneration", c.TransmitterGeneration).Exists().OneOf(DeviceTransmitterGenerations()...)
+	validator.String("unit", c.Unit).OneOf(CalibrationUnits()...)
 	if c.Unit != nil {
 		switch *c.Unit {
 		case CalibrationUnitMgdL:
 			validator.Float64("value", c.Value).Exists().InRange(CalibrationValueMgdLMinimum, CalibrationValueMgdLMaximum)
```

Comment:
I realize this was existing code, but I guess I'm surprised we allowed a floating point value for what ought to be an integer value (mg/dL).

Reply:
Equally strange that the API doc claim it's an `int32` type. I doubt that's true for endpoint that provides mmol/L.

Reply:
my assumption was for `mmol/L` units usage in the future? or  outside "US" usage